# 04 Interpretation

In [3]:
from pathlib import Path
import pandas as pd

ROOT = Path.cwd().resolve().parent

TAB_DIR = ROOT / "outputs" / "tables"
SUM_DIR = ROOT / "outputs" / "summary"
SUM_DIR.mkdir(parents=True, exist_ok=True)

group_summary = pd.read_csv(TAB_DIR / "02_group_summary.csv")
inference = pd.read_csv(TAB_DIR / "03_inference_summary.csv")

print("Group summary:")
display(group_summary)

print("Inference summary:")
display(inference)

Group summary:


,sad_binary,sad_group,n,current_cigarette_users,current_cigarette_proportion,current_cigarette_percent
0,1,Sad/Hopeless: Yes,3854,1064,0.276077,27.607680
1,0,Sad/Hopeless: No,9320,1508,0.161803,16.180258


Inference summary:


,method,comparison,n_yes,x_yes,p_yes,n_no,x_no,p_no,difference,ci_95,z_statistic,p_value,alpha,decision
0,Two-proportion z-test,p_yes - p_no,3854,1064,0.2761,9320,1508,0.1618,0.1143,"[0.0983, 0.1302]",15.0536,< 0.0001,0.05,Reject H0


## Final Interpretation

The current cigarette use proportion was 0.2761 (27.6%) in the Sad/Hopeless: Yes group and 0.1618 (16.2%) in the Sad/Hopeless: No group. The estimated difference was 0.1143, using p_yes - p_no.

The 95% confidence interval for the difference was **[0.0983, 0.1302]**. Since the entire interval is above 0, the result suggests that the Sad/Hopeless: Yes group had a higher current cigarette use proportion.

The two-proportion z-test gave z = 15.0536 and p-value < 0.0001. At alpha = 0.05, we reject H0. Therefore, there is statistically significant evidence that the current cigarette use proportion differs between the two groups.

This result shows an association, not causation, because the YRBS data are observational survey data.

In [4]:
row = inference.iloc[0]

p_yes = row["p_yes"]
p_no = row["p_no"]
diff = row["difference"]
ci_95 = row["ci_95"]
z = row["z_statistic"]
p_value = row["p_value"]
decision = row["decision"]

final_text = f"""Final Interpretation - Cycle 3 Question 8

The current cigarette use proportion was {p_yes:.4f} ({p_yes * 100:.1f}%) in the Sad/Hopeless: Yes group and {p_no:.4f} ({p_no * 100:.1f}%) in the Sad/Hopeless: No group.

The estimated difference was {diff:.4f}, using p_yes - p_no. This means that the Sad/Hopeless: Yes group had a higher current cigarette use proportion than the Sad/Hopeless: No group.

The 95% confidence interval for the difference was {ci_95}. Since the entire interval is above 0, the result suggests that the Sad/Hopeless: Yes group had a higher current cigarette use proportion.

The two-proportion z-test gave z = {z:.4f} and p-value {p_value}. At alpha = 0.05, we {decision}. Therefore, there is statistically significant evidence that the current cigarette use proportion differs between the two groups.

This result shows an association, not causation, because the YRBS data are observational survey data.
"""

(SUM_DIR / "final_interpretation.txt").write_text(final_text, encoding="utf-8")

898